src/components/ReportActionItem/TripRoomPreview.tsx

```diff
@@ -0,0 +1,204 @@
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import Navigation from '@libs/Navigation/Navigation';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as TripReservationUtils from '@libs/TripReservationUtils';
+import type {ContextMenuAnchor} from '@pages/home/report/ContextMenu/ReportActionContextMenu';
+import variables from '@styles/variables';
+import * as Expensicons from '@src/components/Icon/Expensicons';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type {ReportAction} from '@src/types/onyx';
+import type {Reservation} from '@src/types/onyx/Transaction';
+
+type TripRoomPreviewProps = {
+    /** All the data of the action */
+    action: ReportAction;
+
+    /** The associated chatReport */
+    chatReportID: string;
+
+    /** Extra styles to pass to View wrapper */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Popover context menu anchor, used for showing context menu */
+    contextMenuAnchor?: ContextMenuAnchor;
+
+    /** Callback for updating context menu active state, used for showing context menu */
+    checkIfContextMenuActive?: () => void;
+
+    /** Whether a message is a whisper */
+    isWhisper?: boolean;
+
+    /** Whether the corresponding report action item is hovered */
+    isHovered?: boolean;
+};
+
+type ReservationViewProps = {
+    reservation: Reservation;
+};
+
+function ReservationView({reservation}: ReservationViewProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const reservationIcon = TripReservationUtils.getTripReservationIcon(reservation.type);
+    const title = reservation.type === CONST.RESERVATION_TYPE.CAR ? reservation.carInfo?.name : reservation.start.longName;
+
+    const titleComponent =
+        reservation.type === CONST.RESERVATION_TYPE.FLIGHT ? (
+            <View style={[styles.flexRow, styles.alignItemsCenter, styles.gap2]}>
+                <Text style={styles.labelStrong}>{reservation.start.shortName}</Text>
+                <Icon
+                    src={Expensicons.ArrowRightLong}
+                    width={variables.iconSizeSmall}
+                    height={variables.iconSizeSmall}
+                    fill={theme.icon}
+                />
+                <Text style={styles.labelStrong}>{reservation.end.shortName}</Text>
+            </View>
+        ) : (
+            <Text
+                numberOfLines={1}
+                style={styles.labelStrong}
+            >
+                {title}
+            </Text>
+        );
+
+    return (
+        <MenuItemWithTopDescription
+            description={translate(`travel.${reservation.type}`)}
+            descriptionTextStyle={styles.textMicro}
+            titleComponent={titleComponent}
+            titleContainerStyle={styles.gap1}
+            secondaryIcon={reservationIcon}
+            shouldShowRightIcon={false}
```

Comment:
A few of these `should___` probs are false by default and are redundant / could be removed